in_memory_person_storage.go

```diff
@@ -21,9 +21,58 @@ func (s *InMemoryPersonStorage) Add(person *Person) error {
 }
 
 func (s *InMemoryPersonStorage) GetAll() []*Person {
-	persons := make([]*Person, 0)
+	var persons []*Person
 	for _, person := range s.data {
 		persons = append(persons, person)
 	}
 	return persons
 }
+
+func (s *InMemoryPersonStorage) GetPersonById(id uuid.UUID) *Person {
+	p, ok := s.data[id]
+	if !ok {
+		return nil
+	}
+	return p
+}
+
+func (s *InMemoryPersonStorage) GetPersonsByName(name string) []*Person {
+	var persons []*Person
+	for _, val := range s.data {
+		if val.Name == name {
+			persons = append(persons, val)
+		}
+	}
+	return persons
+}
+
+func (s *InMemoryPersonStorage) GetPersonsByCommunication(value string) []*Person {
+	var persons []*Person
+	for _, val := range s.data {
+		for _, comm := range val.Communications {
+			if comm.Value == value {
+				persons = append(persons, val)
+				break
+			}
+		}
+	}
+	return persons
+}
+
+func (s *InMemoryPersonStorage) UpdatePerson(person *Person) bool {
```

Comment:
it's better to return updated and error (*Person, error).
if person is not found we return nil, personNotFoundError